### ISSUE TITLE
Heterogeneous builds are broken

### ISSUE THREAD

**Issue by jsquyres** [open-mpi/ompi#540]
@hjelmn It looks like the latest round of updates broke --enable-heterogeneous, as reported by @siegmargross:

http://www.open-mpi.org/community/lists/users/2015/04/26753.php

When I --enable-heterogeneous on Linux, I get this build error in ob1:

```
  CC       pml_ob1.lo
In file included from pml_ob1.h:38:0,
                 from pml_ob1.c:46:
pml_ob1_hdr.h: In function 'ob1_hdr_ntoh':
pml_ob1_hdr.h:204:12: error: 'mca_pml_ob1_rget_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = ntohl((h).hdr_seg_cnt);       \
            ^
pml_ob1_hdr.h:440:13: note: in expansion of macro 'MCA_PML_OB1_RGET_HDR_NTOH'
             MCA_PML_OB1_RGET_HDR_NTOH(hdr->hdr_rget);
             ^
pml_ob1_hdr.h:204:36: error: 'mca_pml_ob1_rget_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = ntohl((h).hdr_seg_cnt);       \
                                    ^
pml_ob1_hdr.h:440:13: note: in expansion of macro 'MCA_PML_OB1_RGET_HDR_NTOH'
             MCA_PML_OB1_RGET_HDR_NTOH(hdr->hdr_rget);
             ^
pml_ob1_hdr.h:357:12: error: 'mca_pml_ob1_rdma_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = ntohl((h).hdr_seg_cnt);          \
            ^
pml_ob1_hdr.h:449:13: note: in expansion of macro 'MCA_PML_OB1_RDMA_HDR_NTOH'
             MCA_PML_OB1_RDMA_HDR_NTOH(hdr->hdr_rdma);
             ^
pml_ob1_hdr.h:357:36: error: 'mca_pml_ob1_rdma_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = ntohl((h).hdr_seg_cnt);          \
                                    ^
pml_ob1_hdr.h:449:13: note: in expansion of macro 'MCA_PML_OB1_RDMA_HDR_NTOH'
             MCA_PML_OB1_RDMA_HDR_NTOH(hdr->hdr_rdma);
             ^
pml_ob1_hdr.h: In function 'ob1_hdr_hton_intr':
pml_ob1_hdr.h:211:12: error: 'mca_pml_ob1_rget_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = htonl((h).hdr_seg_cnt);       \
            ^
pml_ob1_hdr.h:486:13: note: in expansion of macro 'MCA_PML_OB1_RGET_HDR_HTON'
             MCA_PML_OB1_RGET_HDR_HTON(hdr->hdr_rget);
             ^
pml_ob1_hdr.h:211:36: error: 'mca_pml_ob1_rget_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = htonl((h).hdr_seg_cnt);       \
                                    ^
pml_ob1_hdr.h:486:13: note: in expansion of macro 'MCA_PML_OB1_RGET_HDR_HTON'
             MCA_PML_OB1_RGET_HDR_HTON(hdr->hdr_rget);
             ^
pml_ob1_hdr.h:366:12: error: 'mca_pml_ob1_rdma_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = htonl((h).hdr_seg_cnt);          \
            ^
pml_ob1_hdr.h:495:13: note: in expansion of macro 'MCA_PML_OB1_RDMA_HDR_HTON'
             MCA_PML_OB1_RDMA_HDR_HTON(hdr->hdr_rdma);
             ^
pml_ob1_hdr.h:366:36: error: 'mca_pml_ob1_rdma_hdr_t' has no member named 'hdr_seg_cnt'
         (h).hdr_seg_cnt = htonl((h).hdr_seg_cnt);          \
                                    ^
pml_ob1_hdr.h:495:13: note: in expansion of macro 'MCA_PML_OB1_RDMA_HDR_HTON'
             MCA_PML_OB1_RDMA_HDR_HTON(hdr->hdr_rdma);
             ^
pml_ob1.c: In function 'mca_pml_ob1_send_fin':
pml_ob1.c:665:18: error: 'hdr' undeclared (first use in this function)
     ob1_hdr_hton(hdr, MCA_PML_OB1_HDR_TYPE_FIN, proc);
                  ^
pml_ob1_hdr.h:465:43: note: in definition of macro 'ob1_hdr_hton'
     ob1_hdr_hton_intr((mca_pml_ob1_hdr_t*)h, t, p)
                                           ^
pml_ob1.c:665:18: note: each undeclared identifier is reported only once for each function it appears in
     ob1_hdr_hton(hdr, MCA_PML_OB1_HDR_TYPE_FIN, proc);
                  ^
pml_ob1_hdr.h:465:43: note: in definition of macro 'ob1_hdr_hton'
     ob1_hdr_hton_intr((mca_pml_ob1_hdr_t*)h, t, p)
                                           ^
make[2]: *** [pml_ob1.lo] Error 1
make[2]: Leaving directory `/home/jsquyres/git/ompi/ompi/mca/pml/ob1'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/jsquyres/git/ompi/ompi'
make: *** [all-recursive] Error 1
```

Probably pretty simple to fix -- can you have a look?

**Comment by ggouaillardet** [comment]
looks like a duplicate of #403 
I will work on that this week

**Comment by jsquyres** [comment]
@ggouaillardet Sweet; thanks.

**Comment by siegmargross** [comment]
Great !! I can build everything once more. Thank you very much to everybody who helped to fix the problem.
